src/App.tsx

```diff
@@ -1,15 +1,39 @@
-import React from 'react';
+import { debounce } from 'lodash';
+import React, { useCallback, useMemo, useState } from 'react';
+
 import './App.scss';
+import { Autocomplete } from './components/Autocomplete';
+
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [value, setValue] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [focus, setFocus] = useState(false);
+  const [selectedUser, setSelectedUser] = useState<Person | null>(null);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 5300), []);
```

Comment:
The debounce delay is hardcoded to 5300ms. According to the task description, the delay should be customizable via props with a default value of 300ms. Consider refactoring to allow the delay to be passed as a prop.